### DISCUSSION TITLE
A few timing number

### DISCUSSION THREAD

**Discussion by cecilehannay** [NCAR/amwg_dev#160] [X: General]
I often get asked about timing and throughput. Here are a few numbers:

**Coupled** 

L58 (B1850WscMOM + cice 5)
Model Cost:            6135.95   pe-hrs/simulated_year
Model Throughput:         8.73   simulated_years/day 
Number of PE = 2232 (See pe_layout at #158)  

**Standalone**

L58 (FWscHIST)
Model Cost:            5412.36   pe-hrs/simulated_year 
Model Throughput:         9.58   simulated_years/day 
Number of PE =  2160 (See pe_layout at #152) 

L93 (FWscHIST)
Model Cost:           17931.19   pe-hrs/simulated_year 
Model Throughput:         5.78   simulated_years/day 
Number of PE = 4320 (See pe_layout at #40)

@JulioTBacmeister

**Comment by cecilehannay** [comment]
I added the total number of processors for each run, as it help to interpret the numbers.

**Comment by adamrher** [comment]
how much i/o is in these runs?

**Reply by adamrher** [reply]
we should run with ```./xmlchange TIMER_LEVEL=10``` then I can run my pie chart timing plot that will break it down in terms of i/o, physics, dynamics. Can we do this?

**Reply by adamrher** [reply]
ahh .. it's the time-stepping in L93:

```
se_nsplit=3
se_rsplit=6
se_hypervis_subcycle=3
se_hypervis_subcycle_sponge =  1
```

**Reply by PeterHjortLauritzen** [reply]
Just to add some more context:

The L58 model has a top around 42km and the time-steps are scaled for a max wind speed of 120m/s (CSLAM runs with a 900s time-step)
The L93 model has a top around 80km and the time-steps are scaled for a max wind of over 300m/s (CSLAM runs with a 600s time-step)

Hence tracer transport alone is about 33% more expensive due to the shorter time-step. 

For WACCM with a 140km top the time-steps must be reduced further since the max winds in simulations is roughly 400m/s.

It is correct that cost scales linearly with number of levels if the model top remains the same. If the model top is moved up where large winds occur the scaling changes.

These results are independent of dycore. Here is an example of MPAS-WACCM (70 levels) showing maximum wind speed in the domain every dynamics time-step:

![Screen Shot 2022-08-17 at 5 25 44 PM](https://user-images.githubusercontent.com/8782989/185260695-ee30c2a2-0882-46d7-9855-dfca3edbab48.png)
[Data courtesy of Soudeh Kamali, HAO, NCAR; the zero points are spurious - appears from restarting model]

The plot below (just for the interested reader) is the level/height at which the max/min winds occur:
![Screen Shot 2022-08-17 at 5 27 15 PM](https://user-images.githubusercontent.com/8782989/185260911-acbf72ef-2d17-49a2-8ee9-74a5123fc515.png)
